tests/postgresOpsTests.js

```diff
@@ -0,0 +1,265 @@
+    } catch (error) {
+        jqUnit.assertEquals("Check SELECT from non-existent table", error.message, "relation \"nosuchtable\" does not exist");
+    }
+
+    // Test retrieval of a JSONB value
+    response = await postgresHandler.runSql("SELECT \"colourMap\" FROM rgb WHERE id='chartreuse'");
+    jqUnit.assertNotEquals("Check for empty result", 0, response.length);
+    jqUnit.assertDeepEq("Check value retrieved", rgbChartreuse.colourMap, response.rows[0].colourMap);
+
+    // Test failing case where column does not exist
+    try {
+        await postgresHandler.runSql("SELECT \"noSuchColumn\" FROM rgb WHERE color='chartreuse';");
+    } catch (error) {
+        jqUnit.assertEquals("Check SELECT from non-existent column", error.message, "column \"noSuchColumn\" does not exist");
+    }
+
+    // Insert new user record
+    response = await postgresHandler.runSql(testData.userToInsert);
+    const expectedKeys = Object.keys(testData.userToInsertJSON);
+    fluid.tests.utils.checkKeyValuePairs(expectedKeys, response.rows[0], testData.userToInsertJSON, "Check added record");
+
+    // Insert new record again -- should fail
+    try {
+        await postgresHandler.runSql(testData.userToInsert);
+    } catch (error) {
+        jqUnit.assertEquals("Check second INSERT of same record", error.message, "duplicate key value violates unique constraint \"users_pkey\"");
+    }
+
+    // Update a field with a proper identifier
+    response = await postgresHandler.runSql(testData.userChanges);
+    fluid.tests.utils.checkKeyValuePairs(Object.keys(expected.userChanges), response.rows[0], expected.userChanges, "Check update results");
+
+    // Update the primary key itself
+    response = await postgresHandler.runSql(testData.primaryKeyChange);
+    fluid.tests.utils.checkKeyValuePairs(Object.keys(expected.primaryKeyChange), response.rows[0], expected.primaryKeyChange, "Check update results");
+
+    // Update with non-existent primary key; should return zero results
+    response = await postgresHandler.runSql("UPDATE users SET iterations=55 WHERE \"userId\"='noSuchUser';");
+    jqUnit.assertEquals("Check UPDATE with mismatched primaryKey", response.rowCount, 0);
+
+    // Test successful deletion
+    response = await postgresHandler.runSql("DELETE FROM \"users\" WHERE \"userId\"='some.new.id';");
+    // response = await postgresHandler.runSql("DELETE FROM \"prefsSafes\" WHERE \"prefsSafesId\"='prefsSafe-1';");
+    jqUnit.assertEquals("Check number of records deleted", response.rowCount, 1);
+
+    // Run sql from a file.
+    response = await postgresHandler.runSqlFile(sqlFile);
+    fluid.tests.utils.testResults(response, 3);
+
+    // Run sql from a non-existent file -- should fail
+    try {
+        await postgresHandler.runSqlFile("/no/such/file.sql");
+    } catch (error) {
+        jqUnit.assertEquals("Check running sql when file access failure", error.message, "ENOENT: no such file or directory, open '/no/such/file.sql'");
```

Comment:
The wording of this error is slightly different on Windows - "ENOENT: no such file or directory, open 'e:\no\such\file.sql'" - this can be replaced with a slightly weaker check that the beginning code is correct and that the filename appears somewhere in the path.

Also note that this and several previous assertions have "expected" and "actual" flipped around